lib/viewmodels/posts_feed_view_model.dart

```diff
@@ -46,13 +46,13 @@ class PostsFeedViewModel extends AutoDisposeAsyncNotifier<List<PostDetails>> {
       position,
     );
 
-    // Wait for both futures to complete for optimisation
-    final List<dynamic> results = await Future.wait([
+    final results = await Future.wait([
       postsFirestoreFuture,
       challengesFuture,
     ]);
-    List<PostFirestore> postsFirestore = results[0];
-    final List<ChallengeFirestore> challenges = results[1];
+    List<PostFirestore> postsFirestore = results[0] as List<PostFirestore>;
```

Comment:
On those two lines the code doesn't need to specify the type of the variable twice.

You can just have `final varName = ... as Type;`

Reply:
I fixed one line with [e1d5922](https://github.com/ProximaEPFL/proxima/pull/262/commits/e1d5922b9b2629d9ee45df6a4fdfb8ee016550bf), but for the other one I don't think that I can remove the type specification as this variable can't be final (it's modified later in the code).

Reply:
then you can use `var` instead of `final`